.github/workflows/ci.yml

```diff
@@ -23,26 +23,24 @@ jobs:
     steps:
       - run: exit 0
 
+  # Job to filter crates to be published
   gather_crates:
     runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
     outputs:
-      members: ${{ steps.cargo-metadata.outputs.members }}
+      crates: ${{ steps.filter.outputs.changes }}
     steps:
-      - uses: actions/checkout@v3
-
-      - id: cargo-metadata
-        run: |
-          WORKSPACE_MEMBERS=$(cargo metadata --format-version=1 --no-deps | jq -c '.packages | .[] | select(.publish == null) | .name' | jq -s '.' | jq -c '.')
-          echo "members=${WORKSPACE_MEMBERS}" >> $GITHUB_OUTPUT
+      - uses: dorny/paths-filter@v2
```

Comment:
TIL, this is very nice Philippe thanks! Very clean way to detect the changes to `Cargo.toml` files and only published the changed ones, CC @tchataigner.